### PR TITLE
Use the OpenSSL or GnuTLS base64 functions

### PR DESCRIPTION
**Pull request by dfandrich**
These should be better optimized than the generic versions bundled
with inadyn, and reduce the size of the final binary.

### CODE REVIEW

**Comment by troglobit** [comment]
Inadyn can be built without a SSL/TLS library.  If you want this merged I suggest adding checks for that so we don't break that usecase.  Also, this patch does not follow the coding style of the project.

**Comment by dfandrich** [comment]
I've retained support for compiling without TLS and changed the style to hopefully better conform.

**Comment by troglobit** [comment]
This looks better, thanks!  For future reference, avoid C++ style comments in C projects.

**Comment by dfandrich** [comment]
On Tue, Sep 14, 2021 at 07:49:30AM -0700, Joachim Wiberg wrote:
> This looks better, thanks! For future reference, avoid C++ style comments in C
> projects.

// comments have been part of C for over 20 years :-)

**Comment by troglobit** [comment]
Fine, let me rephrase. Please avoid C++ style comments in this project.
